### PR TITLE
Migrate docker build from DockerHub to GitHub Container Registry

### PR DESCRIPTION
**Pull request by eastata**
The issue is a part of global migration from DockerHub to [GitHub Container Registry](https://docs.github.com/en/packages/guides/about-github-container-registry).

* Docker images for applications **mpc**, **client**, **prover**
  * builded by buildx for `linux/amd64`. List of platform can be extended for the future
  * building use contaner registry as a remote docker cache cache-mpc, cache-client, cache-prover
  * initial build generate docker images with tags based on git hash. Workflow is ready for extention by docker-compose for integral tests
  * Build triggered by release tag `vX.Y.Z`

* **Base** image still the same, only migrated to GitHub CR
  * Build triggered by release tag `docker-base-vX.Y.Z`
 

## Docker images
| Docker files | Image | Tags | Description |
|---------------|------|-----|--|
| [./Dockerfile-prover](./Dockerfile-prover) | [ghcr.io/clearmatics/zeth:latest-prover](https://github.com/orgs/clearmatics/packages/container/package/zeth) | `latest-prover`, `X.Y.Z-prover` - Release of zeth, `git-%HASH%-prover` - developers build by git-commit  | [Zeth Prover Server](./prover_server/README.md). Image use `zeth-base` for building |
| [./Dockerfile-client](./Dockerfile-client) | [ghcr.io/clearmatics/zeth:latest-client](https://github.com/orgs/clearmatics/packages/container/package/zeth) | `latest-client`, `X.Y.Z-client` - Release of zeth, `git-%HASH%-client` - developers build by git-commit  | [Python client to interact with the prover](./client/README.md) |
| [./Dockerfile-mpc](./Dockerfile-mpc) | [ghcr.io/clearmatics/zeth:latest-mpc](https://github.com/orgs/clearmatics/packages/container/package/zeth) | `latest-mpc`, `X.Y.Z-mpc` - Release of zeth, `git-%HASH%-mpc` - developers build by git-commit  | [Tools for Multi-Party Computation](./mpc/README.md). Image use `zeth-base` for building |
| [./Dockerfile-base](./Dockerfile-base) | [ghcr.io/clearmatics/zeth:latest-base](https://github.com/orgs/clearmatics/packages/container/package/zeth) | `latest-base`, `A.B.C-base` - Release of zeth-base | Base image for building other containers |

### CODE REVIEW

**Review by dtebbs — APPROVED** [review]
LGTM

File: .github/workflows/dev-docker-apps.yml [hunk 1]
```diff
@@ -32,22 +32,44 @@ jobs:
...
+        run: echo ::set-output name=COMMIT_HASH::$(git rev-parse --short HEAD)
+
+      - name: "Configure Docker"
+        run: echo ${{ secrets.CR_PAT }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
+
+      - name: "Run service"
+        run: echo "Here we will run docker-compose based on images git-"${{ steps.git_vars.outputs.COMMIT_HASH }}
+
+      - name: "Test RPC"
+        run: echo "Here we will run a tests"
```
Comment (AntoineRondelet) [inline .github/workflows/dev-docker-apps.yml:75]: Not sure I get the purpose of this job @eastata 
What are the "tests" doing here, beyond the `echo` messages. You don't seem to do a `docker-compose up` (as suggested in the `Run service` step above, do you?)

File: .github/workflows/release-docker-base.yml [hunk 2]
```diff
@@ -5,27 +5,20 @@ on:
...
         uses: actions/checkout@v2
 
-      - name: "Get vars from git"
-        id: git_vars
-        run: echo ::set-output name=TAG::${GITHUB_REF/refs\/tags\/docker-base-/}
-
       - name: "Create swap from file"
         run: |
-            sudo fallocate -l 21G /swapfile2
+            sudo fallocate -l 19G /swapfile2
```
Comment (AntoineRondelet) [inline .github/workflows/release-docker-base.yml:21]: Out of curiosity, how the range `19G` chosen here?

**Comment by AntoineRondelet** [comment]
>     * Build triggered by release tag `vX.Y.Z`

@eastata - can we still manually push docker images to this registry? I'd like to push the "application images" for the 0.5 and 0.6 tags (which I guess, won't be re-built - or will they?)

File: .github/workflows/dev-docker-apps.yml (hunk 1 above)
Comment (eastata) [reply .github/workflows/dev-docker-apps.yml:75]: Yes. It is placeholder for docker-compose.
I guess the integral tests should be implemented at another issue/PR

File: .github/workflows/release-docker-base.yml (hunk 2 above)
Comment (eastata) [reply .github/workflows/release-docker-base.yml:21]: Not enough space on local disk for images if allocate 21G for swap

**Comment by eastata** [comment]
> can we still manually push docker images to this registry

Yes.  You have to [configure docker client](https://docs.github.com/en/packages/guides/configuring-docker-for-use-with-github-packages#authenticating-to-github-packages) using [GITHUB_TOKEN](https://docs.github.com/en/packages/guides/configuring-docker-for-use-with-github-packages#authenticating-with-the-github_token)

**Comment by AntoineRondelet** [comment]
> > can we still manually push docker images to this registry
> 
> Yes. You have to [configure docker client](https://docs.github.com/en/packages/guides/configuring-docker-for-use-with-github-packages#authenticating-to-github-packages) using [GITHUB_TOKEN](https://docs.github.com/en/packages/guides/configuring-docker-for-use-with-github-packages#authenticating-with-the-github_token)

I see, thanks @eastata

File: .github/workflows/dev-docker-apps.yml (hunk 1 above)
Comment (AntoineRondelet) [reply .github/workflows/dev-docker-apps.yml:75]: I see, thanks @eastata 
Let's comment this job out for now. I think it is not necessary to run an extra job if it only aims at being a placeholder and run `echo`s. We'll see if we need to do extra tests with `docker-compose`, and add them in a later PR if necessary (bearing in mind that @dtebbs added a few test jobs in #326 already)

Comment (eastata) [reply .github/workflows/dev-docker-apps.yml]: Removed

**Review by AntoineRondelet — APPROVED** [review]
LGTM. Thanks @eastata!

**Pull request merged**